docs/content/guides/dagster-pipes.mdx

```diff
@@ -0,0 +1,78 @@
+---
+title: Dagster Pipes | Dagster Docs
+description: "Dagster Pipes provides a protocol between the orchestration environment (Dagster) and external execution (ex: Databricks) and a toolkit for building implementations of that protocol."
+---
+
+# Dagster Pipes
+
+Dagster Pipes is a toolkit for building integrations between Dagster and external execution environments. It standardizes the process of passing parameters, injecting context information, ingesting logs, and collecting metadata all while remaining agnostic to how remote computations are launched in those environments. This enables the separation of orchestration and business logic in the Dagster ecosystem.
+
+It also smooths the process of incorporating pre-existing code, business logic, and execution environments into Dagster. Using Dagster Pipes, you add just a few lines to your code and it becomes seamlessly invocable by the orchestrator. In turn, you can stream logs and metadata back to Dagster so you can leverage its observability, lineage, cataloging, and debugging capabilities.
+
+---
+
+## Benefits
+
+With Dagster Pipes, you can:
+
+- Incorporate existing code into Dagster without huge refactors
+- Onboard stakeholder teams onto Dagster incrementally
+- Run code in external environments, and:
+  - Easily pass parameters to the code
+  - Stream unstructured logs and structured metadata back to Dagster
+- Separate orchestration and business logic environments
+- Use languages other than Python with Dagster
+
+---
+
+## Limitations
+
+While Dagster Pipes is lightweight and flexible, there are a few limitations to be aware of:
+
+- **Step launchers are currently incompatible.**
+- **I/O managers aren’t currently supported.** Dagster Pipes isn't built for processing data in memory, which is what I/O managers are for. For example, to process data in memory in Pyspark, you can use an I/O manager [as demonstrated in the Pyspark integration guide](/integrations/spark#running-pyspark-code-in-assets). Otherwise, in a remote case, you can use Pipes.
+
+---
+
+## How it works
+
+Dagster Pipes provides a protocol between the orchestration environment (Dagster) and external execution (ex: Databricks) and a toolkit for building implementations of that protocol.
+
+When Dagster Pipes is invoked, several steps will be carried out both in the **orchestration process**, or Dagster, and in the **external process**, such as Databricks.
+
+### In the orchestration process (Dagster)
+
+When the execution of a Dagster asset begins, Dagster makes context information (ex: `partition_key`, `asset_key`, etc.) available for the external process to consume.
+
+Next, Dagster launches the external process and passes in any provided parameters.
```

Comment:
Wonder if this can be even more high level like:

"When Dagster Pipes is called in a Dagster asset, Dagster launches the external process with parameters and context information (ex: `partition_key`, `asset_key`, etc.) "

Reply:
That's fine by me